spec/models/game_spec.rb

```diff
@@ -381,10 +382,200 @@ def expect_a_piece(attributes)
             # data setup
             create(:king, player: game.player_2, game: game, x_coordinate: 4, y_coordinate: 1, color: 'black')
 
-            expect(game.check?).to be true
+            expect(game.check?(opposite_color)).to be true
           end
         end
       end
     end
   end
+
+  describe '#checkmate?' do
+    let(:game) { create(:game) }
+    let!(:player1_king) { create(:king, player: game.player_1, game: game, x_coordinate: 0, y_coordinate: 0, color: 'white', captured: false) }
+    let!(:player2_king) { create(:king, player: game.player_2, game: game, x_coordinate: 4, y_coordinate: 4, color: 'black', captured: false) }
+
+    let(:opposite_color) { 'black' }
+
+    context 'the king can run away' do
+      let!(:player2_bishop) { create(:bishop, player: game.player_2, game: game, x_coordinate: 2, y_coordinate: 2, color: 'black', captured: false) }
+      # _______________
+      # |__|__|__|__|BK| 4
+      # |__|__|__|__|__| 3
+      # |__|__|BB|__|__| 2
+      # |__|__|__|__|__| 1
+      # |wk|__|__|__|__| 0
+      # | 0  1  2  3  4
+      it 'should returns false' do
+        expect(game.checkmate?(opposite_color)).to be false
+        expect(player1_king.x_coordinate).to be 0
+        expect(player1_king.y_coordinate).to be 0
+      end
+    end
+
+    context 'the king cannot run away' do
+      let!(:player2_rook) { create(:rook, player: game.player_2, game: game, x_coordinate: 4, y_coordinate: 0, color: 'black', captured: false) }
+      let!(:player2_rook) { create(:rook, player: game.player_2, game: game, x_coordinate: 0, y_coordinate: 4, color: 'black', captured: false) }
```

Comment:
you have two variables with the same name (`player2_rook`) here, the second one will override the first one, and you end up with only one black rook at location (0, 4).